packages/runtime/inferno-hooks/recorder.ts

```diff
@@ -79,15 +95,19 @@ export function createRecorder(component: HookContainer) {
       nextState: any,
       context: any,
     ) {
-      shouldUpdate = !equal(component.props.renderProps, nextProps.renderProps);
+      let shouldUpdate = true;
+      if (component.props.pure) {
+        const propsEqual = equal(component.props.renderProps, nextProps.renderProps);
+        if (propsEqual) {
+          shouldUpdate = !equal(component.state, nextState);
+        }
+      }
```

Comment:
I guess a right part will never be executed if the left is false
Better keep readability

```suggestion
      const shouldUpdate = !(component.props.pure &&
        equal(component.props.renderProps, nextProps.renderProps) &&
        equal(component.state, nextState)
      );
```